pkg/controller/perconaxtradbcluster/perconaxtradbcluster_controller.go

```diff
@@ -333,6 +339,48 @@ func (r *ReconcilePerconaXtraDBCluster) deploy(cr *api.PerconaXtraDBCluster) err
 	return nil
 }
 
+func (r *ReconcilePerconaXtraDBCluster) checkSecret(cr *api.PerconaXtraDBCluster, namespace string) error {
+	secretObj := corev1.Secret{}
+	err := r.client.Get(context.TODO(),
+		types.NamespacedName{
+			Namespace: namespace,
+			Name:      cr.Spec.PXC.SSLSecretName,
+		},
+		&secretObj)
+	if err != nil {
+		return err
```

Comment:
Add more context to the error. Here and below

Reply:
Execution never go further if there is no secret. In that case `Get` will return `StatusReasonNotFound` error.